ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/AbstractMaskingProvider.java

```diff
@@ -122,4 +141,43 @@ public void setName(String name) {
   public String getName() {
     return name;
   }
+
+  protected String applyUnexpectedValueHandling(String input, Supplier<String> randomGenerator) {
+    debugFaultyInput(getName());
+    String response;
+
+    if (unexpectedInputHandler != null) {
+      switch (unexpectedInputHandler) {
+        case NULL:
+          response = null;
+          break;
+        case RANDOM:
+          response = randomGenerator == null ? null : randomGenerator.get();
+          break;
+        case MESSAGE:
+          response = unexpectedInputReturnMessage;
+          break;
+        case ERROR_EXIT:
+          throw new PrivacyProviderInvalidInputException(input, getName());
+        default:
+          response = null;
+      }
+    } else {
+      if (unspecifiedValueHandling == 2) {

```

Comment:
Should we create constants and use those everywhere and get rid of the digits ? I see that the digits are concentrated all into this method at least so it's not as bad as before.

Reply:
I will create constants for the old-style numeric values as part of the next delivery for this story.